packages/lit-dev-content/src/github/github-util.ts

```diff
@@ -8,3 +8,59 @@
  * Specifies the GitHub API version we expect.
  */
 export const JSON_GITHUB_API_V3 = 'application/vnd.github.v3+json';
+
+/**
+ * An error encountered while querying the GitHub API.
+ */
+export class GitHubError extends Error {
+  readonly url: URL;
+  readonly status?: number;
+
+  constructor(
+    url: URL,
+    method: string,
+    status: number | undefined,
+    message?: string,
+    stack?: string
+  ) {
+    super(
+      `Error calling GitHub API ${method} ${url.pathname}` +
+        (status ? ` (HTTP status ${status})` : '') +
+        (message ? `: ${message}` : '')
+    );
+    this.url = url;
+    this.status = status;
+    if (stack) {
+      this.stack = stack;
+    }
+  }
+}
+
+/**
+ * Perform a fetch and return the JSON response. Throw on non-2XX HTTP status
```

Comment:
Is it possible to get a valid 3XX status here?

Edit: Ah I see that the check is if `res.ok`.